### PR TITLE
Improve tests on travis

### PR DESCRIPTION
**Pull request by chregu**
- test also on go 1.11
- do linting only on go 1.11
- do linting in a seperate job

### CODE REVIEW

File: .travis/lint.sh [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+# linting
+
+go get -u honnef.co/go/tools/cmd/megacheck
+go get -u golang.org/x/lint/golint
+go get -u golang.org/x/tools/cmd/goimports
+
+# go 1.8 doesn't exclude /vendor/ yet.
+dirs=$(go list -f {{.Dir}} ./... | grep -v /vendor/)
+pkgs=$(go list ./... | grep -v /vendor/)
```
Comment (mweibel) [inline .travis/lint.sh]: I assume this could go away now

File: .travis.yml [hunk 2]
```diff
@@ -30,6 +27,9 @@ stages:
 
 jobs:
   include:
+    - name: "Linting with Go: 1.10"
+      script: ./.travis/lint.sh
+      go: "1.10"
```
Comment (mweibel) [inline .travis.yml]: let's do the deploy and lint in the latest version (1.11), no?
```suggestion
      go: "1.11"
```

File: .travis.yml [hunk 3]
```diff
@@ -30,6 +27,9 @@ stages:
...
     - stage: deploy
       script: ./.travis/deploy.sh
       go: "1.10"
```
Comment (mweibel) [inline .travis.yml]:
```suggestion
      go: "1.11"
```

File: .travis/lint.sh (hunk 1 above)
Comment (chregu) [reply .travis/lint.sh]: you mean the `grep -v /vendor/` part? Looks like

Comment (mweibel) [reply .travis/lint.sh]: exactly.

Comment (chregu) [reply .travis/lint.sh]: removed

File: .travis.yml (hunk 2 above)
Comment (chregu) [reply .travis.yml]: done

File: .travis.yml (hunk 3 above)
Comment (chregu) [reply .travis.yml]: done

**Pull request merged**